cadquery/occ_impl/importers/assembly.py

```diff
@@ -0,0 +1,198 @@
+from OCP.TCollection import TCollection_ExtendedString
+from OCP.Quantity import Quantity_Color, Quantity_ColorRGBA
+from OCP.TDocStd import TDocStd_Document
+from OCP.IFSelect import IFSelect_RetDone
+from OCP.STEPCAFControl import STEPCAFControl_Reader
+from OCP.XCAFDoc import (
+    XCAFDoc_DocumentTool,
+    XCAFDoc_ColorGen,
+    XCAFDoc_ColorSurf,
+    XCAFDoc_GraphNode,
+)
+from OCP.TDF import TDF_Label, TDF_LabelSequence, TDF_AttributeIterator, TDF_DataSet
+from OCP.TDataStd import TDataStd_Name
+
+import cadquery as cq
+from ..assembly import AssemblyProtocol
+
+
+def importStep(assy: AssemblyProtocol, path: str):
+    """
+    Import a step file into an assembly.
+
+    :param assy: An Assembly object that will be packed with the contents of the STEP file.
+    :param path: Path and filename to the STEP file to read.
+
+    :return: None
+    """
+
+    # Document that the step file will be read into
+    doc = TDocStd_Document(TCollection_ExtendedString("XmlOcaf"))
+
+    # Create and configure a STEP reader
+    step_reader = STEPCAFControl_Reader()
+    step_reader.SetColorMode(True)
+    step_reader.SetNameMode(True)
+    step_reader.SetLayerMode(True)
+    step_reader.SetSHUOMode(True)
+
+    # Read the STEP file
+    status = step_reader.ReadFile(path)
+    if status != IFSelect_RetDone:
+        raise ValueError(f"Error reading STEP file: {path}")
+
+    # Transfer the contents of the STEP file to the document
+    step_reader.Transfer(doc)
+
+    # Shape and color tools for extracting XCAF data
+    shape_tool = XCAFDoc_DocumentTool.ShapeTool_s(doc.Main())
+    color_tool = XCAFDoc_DocumentTool.ColorTool_s(doc.Main())
+    layer_tool = XCAFDoc_DocumentTool.LayerTool_s(doc.Main())
+
+    def _process_simple_shape(label, parent_location=None, parent_name=None):
+        shape = shape_tool.GetShape_s(label)
+
+        # Tracks the RGB color value and whether or not it was found
+        cq_color = None
```

Comment:
It does not make sense to initialize here, it is done below. Typing wise this is confusing. What was the rationale?